sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -228,4 +220,101 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("load API") {
+    spark.read.format("org.apache.spark.sql.test").load()
+    spark.read.format("org.apache.spark.sql.test").load(input)
+    spark.read.format("org.apache.spark.sql.test").load(input, input, input)
+    spark.read.format("org.apache.spark.sql.test").load(Seq(input, input): _*)
+    Option(input).map(spark.read.format("org.apache.spark.sql.test").load)
+  }
+
+  test("text - API and common behavior") {
+    // Reader, without user specified schema
+    assert(spark.read.text().schema === textSchema)
+    assert(spark.read.text(input).schema === textSchema)
+    assert(spark.read.text(input, input, input).schema === textSchema)
+    assert(spark.read.text(Seq(input, input): _*).schema === textSchema)
+    assert(Option(input).map(spark.read.text).get.schema === textSchema) // SPARK-16009
+
+    // Reader, with user specified schema
+    assert(spark.read.schema(userSchema).text().schema === userSchema)
```

Comment:
In that case, then its probably better for stuff to fail earlier, that is TextFileFormat should fail if there is a userSpecifiedSchema.